### PR TITLE
Outfits

### PR DESCRIPTION
**Pull request by biotech77**
# Pull Request Description

Outfits Shape

## What does this PR change?

Implements #5617 

### Prerequisites
- [ ] List any required setup steps
- [ ] Include environment/configuration requirements

### Test Steps
1. First step
2. Second step
3. Expected result after step 2
4. ...

### Additional Testing Notes
- Note any edge cases to verify
- Mention specific areas that need careful testing
- List known limitations or potential issues

## Quality Checklist
- [ ] Changes have been tested locally
- [ ] Documentation has been updated (if required)
- [ ] Performance impact has been considered
- [ ] For SDK features: Test scene is included

## Code Review Reference
Please review our [Code Review Standards](https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md) before submitting.

### CODE REVIEW

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Commands/CheckOutfitEquippedStateCommand.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using Cysharp.Threading.Tasks;
+using DCL.AvatarRendering.Wearables.Equipped;
+using DCL.AvatarRendering.Wearables.Helpers;
+using DCL.Backpack.AvatarSection.Outfits.Models;
+using DCL.Backpack.Outfits.Extensions;
+using DCL.Profiles.Self;
+
+namespace DCL.Backpack.AvatarSection.Outfits.Commands
+{
+    public class CheckOutfitEquippedStateCommand
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Commands/CheckOutfitEquippedStateCommand.cs]: This command is not used either?

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Commands/CheckOutfitEquippedStateCommand.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                savedOutfit.outfit.eyes.color != eyesColor ||
+                savedOutfit.outfit.skin.color != skinColor)
+            {
+                return false;
+            }
+
+            var profile = await selfProfile.ProfileAsync(ct);
+            if (profile == null) return false;
+
+            var liveWearableUrns = new HashSet<string>(
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Commands/CheckOutfitEquippedStateCommand.cs]: If this command is really needed please pull the collections

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Commands/PreviewOutfitCommand.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        public void Commit()
+        {
+            originalOutfit = null;
+        }
+
+        private async UniTask<Outfit> CreateOutfitFromEquippedAsync(CancellationToken ct)
+        {
+            var profile = await selfProfile.ProfileAsync(ct);
+            var (hair, eyes, skin) = equippedWearables.GetColors();
+            var bodyShape = equippedWearables.Items().FirstOrDefault(i => i.Key == WearableCategories.Categories.BODY_SHAPE).Value?.GetUrn() ?? "";
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Commands/PreviewOutfitCommand.cs]: It's a dictionary, get an item by key directly

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Commands/PrewarmWearablesCacheCommand.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        private readonly IWearablesProvider wearablesProvider;
+
+        public PrewarmWearablesCacheCommand(IWearablesProvider wearablesProvider)
+        {
+            this.wearablesProvider = wearablesProvider;
+        }
+
+        public async UniTask ExecuteAsync(IReadOnlyCollection<URN> wearableUrns, CancellationToken ct)
+        {
+            if (wearableUrns == null || wearableUrns.Count == 0)
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Commands/PrewarmWearablesCacheCommand.cs]: Respect nullability

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Repository/OutfitsRepository.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+{
+    using OutfitsEntity = EntityDefinitionGeneric<OutfitsMetadata>;
+
+    /// <summary>
+    ///     Handles the deployment of user outfits to the Catalyst network by constructing
+    ///     and publishing an Outfits entity, mirroring the logic of RealmProfileRepository.
+    /// </summary>
+    public class OutfitsRepository
+    {
+        private static readonly JsonSerializerSettings SERIALIZER_SETTINGS =
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Repository/OutfitsRepository.cs]: Not used

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+                // Use Graphics.Blit to perform a GPU-accelerated copy and scale operation.
+                // This is the fastest way to downscale a texture.
+                Graphics.Blit(source, tempRT);
+
+                // Create the final Texture2D with the correct small dimensions.
+                var screenshotTexture = new Texture2D(targetWidth, targetHeight, TextureFormat.RGBA32, false);
+
+                // Set the temporary RT as active and read its (now small) contents.
+                RenderTexture.active = tempRT;
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs]: You can use `Graphics.CopyTexture()` directly instead of setting the render target and calling `ReadPixels`

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+                screenshotTexture.ReadPixels(new Rect(0, 0, targetWidth, targetHeight), 0, 0);
+
+                var pixels = screenshotTexture.GetPixels();
+                for (int p = 0; p < pixels.Length; p++)
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs]: This is super slow! I believe @dalkia and @GBirch33 also needed to apply gamma correction, can you ask them how to do it on GPU?

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+                
+                screenshotTexture.Apply();
+
+                // Clean up: release the temporary RT and reset the active one.
+                RenderTexture.active = null;
+                RenderTexture.ReleaseTemporary(tempRT);
+
+                ct.ThrowIfCancellationRequested();
+
+                byte[] pngBytes = screenshotTexture.EncodeToPNG();
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs]: I believe you can even avoid manipulating pixels in a managed form: you can get a pixel native array and pass it to https://docs.unity3d.com/ScriptReference/ImageConversion.EncodeNativeArrayToPNG.html

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Slots/OutfitSlotPresenter.cs [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+        public override string ToString()
+        {
+            return $"BodyShape: {BodyShapeUrn}, Wearables: [{string.Join(", ", WearableUrns)}], " +
+                   $"SkinColor: {SkinColor}, HairColor: {HairColor}, EyesColor: {EyesColor}, ThumbnailUrl: {ThumbnailUrl}";
+        }
+    }
+
+    public enum OutfitSlotState { Empty, Loading, Full, Save, EmptyAndReadyToSave }
+
+    public class OutfitSlotPresenter : IDisposable
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Slots/OutfitSlotPresenter.cs]: Please respect nullability

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/OutfitsPresenter.cs [hunk 10]
```diff
@@ -0,0 +1,434 @@
...
+        private async UniTaskVoid OnSaveOutfitRequestedAsync(int slotIndex, CancellationToken ct)
+        {
+            var presenter = slotPresenters.FirstOrDefault(p => p.slotIndex == slotIndex);
+            if (presenter == null) return;
+
+            presenter.SetSaving();
+
+            try
+            {
+                var savedItem = await saveOutfitCommand.ExecuteAsync(slotIndex,
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/OutfitsPresenter.cs]: And here

File: Explorer/Assets/DCL/Backpack/CharacterPreview/BackpackCharacterPreviewController.cs [hunk 11]
```diff
@@ -178,6 +180,27 @@ private void OnEmoteSlotSelected(int slot)
             PlayEmote(emote.GetUrn().Shorten());
         }
 
+        public void PlayRandomEmote()
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/CharacterPreview/BackpackCharacterPreviewController.cs]: Can be removed

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs [hunk 12]
```diff
@@ -30,65 +39,73 @@ public async UniTask<Texture2D> TakeAndSaveScreenshotAsync(CharacterPreviewContr
...
+                var srgbNative = req.GetData<byte>();
 
-                ct.ThrowIfCancellationRequested();
+                byte[] pngBytes;
+                // Encode PNG directly from the readback buffer
+                // (no Texture2D.Get/SetPixels, no CPU gamma conversion)
+                using (var pngNative = ImageConversion.EncodeNativeArrayToPNG(
+                           srgbNative, downscaledSRGB.graphicsFormat, (uint)w, (uint)h))
+                {
+                    pngBytes = pngNative.ToArray();
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs]: Don't allocate an array, File has an API that accepts `Span`:
```
var file = File.OpenWrite(filePath);
file.Write(pngNative);
```

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs [hunk 13]
```diff
@@ -30,65 +39,73 @@ public async UniTask<Texture2D> TakeAndSaveScreenshotAsync(CharacterPreviewContr
...
-                else return null;
-
+                await UniTask.SwitchToThreadPool();
                 await File.WriteAllBytesAsync(filePath, pngBytes, ct);
+                await UniTask.SwitchToMainThread(ct);
 
-                return screenshotTexture;
+                // Return bytes so UI can display thumbnail
+                // immediately (no disk read required)
+                return pngBytes;
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Services/AvatarScreenshotService.cs]: I suppose you can use the native array directly, moreover if it's just to create texture you can use `Graphics.Copy` from `downscaledSRGB` directly to the 2d texture without a need to decode again
